lib/util/util.c

```diff
@@ -889,26 +890,33 @@ char * util_alloc_realpath__(const char * input_path) {
     }
 
     {
-      int path_index = 1;  // Path can not start with ..
-      int prev_index = 0;
+      int path_index = 0;
+      int prev_index = -1;
+      int depth = 0;
+
       while (true) {
         if (path_index == path_len)
           break;
 
         if (strcmp(path_list[path_index] , BACKREF ) == 0) {
           mask[path_index] = false;
-          mask[prev_index] = false;
-          prev_index--;
           path_index++;
+          if (depth > 0) {
+            mask[prev_index] = false;
+            prev_index--;
```

Comment:
I believe there is a bug in this function (see #239). But it was here before this PR as well, so I just created an issue for it.

Reply:
Yes - you are right; thank you for the thorough review.